include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -146,8 +146,317 @@ namespace Differentiation
+     *         not a symbolic value or expression.
+     *         An exceptional case is, for example, when performing symbolic
+     *         assembly on a finite element level. When extracting the symbolic
+     *         equivalent of the shape function gradients using FEExtractors,
+     *         the returned tensor will have some <em>a priori</em> determined
+     *         zero-valued components. These trivial components are not valid
+     *         symbols (as they are not symbolic expressions), and we would
+     *         typically wish to guard against their (erroneous) inclusion. In
+     *         this scenario, for convenience, one could set
+     *         @p ignore_invalid_symbols to <tt>true</tt> and these zero-valued
+     *         entries would be skipped over and ignored.
+     *
+     * @note In this function, the @p ValueType is somewhat arbitrary as
+     * it is only used to create dummy values.
+     */
+    template <bool ignore_invalid_symbols = false, typename ValueType = double>
+    void
+    add_to_symbol_map(types::substitution_map &symbol_map,
+                      const Expression &       symbol);
+
+    /**
+     * A convenience function for an adding empty entry, with the key value
+     * given by @p symbol, to the symbolic map @p substitution_map.
+     *
+     * For more context which this function is used, see the other
+     * \ref add_to_symbol_map(types::substitution_map &,const
+     * Expression &) function.
+     *
+     * @tparam ignore_invalid_symbols See the other
+     * \ref add_to_symbol_map(types::substitution_map &,const
+     * Expression &) function for a detailed discussion on the role of this
+     * template argument.
+     *
+     * @note In this function, the @p ValueType is somewhat arbitrary as
+     * it is only used to create dummy values.
+     */
+    template <bool ignore_invalid_symbols = false,
+              typename SymbolicType       = SD::Expression,
+              typename ValueType          = double,
+              typename                    = typename std::enable_if<
+                !std::is_base_of<Expression, SymbolicType>::value &&
+                dealii::internal::is_explicitly_convertible<
+                  SymbolicType,
+                  const SymEngine::RCP<const SymEngine::Basic> &>::value>::type>
+    void
+    add_to_symbol_map(types::substitution_map &symbol_map,
+                      const SymbolicType &     symbol);
+
+    /**
+     * A convenience function for adding empty entries, with the key values
+     * equal to the entries in @p symbols, to the symbolic map @p symbol_map.
+     * It is expected that all entries in the input vector @p symbols be
+     * of @p SymbolicType, compatible with the other add_to_symbol_map()
+     * functions.
+     *
+     * For more context which this function is used, see the other
+     * \ref add_to_symbol_map(types::substitution_map &,const
```

Comment:
```suggestion
     * @ref add_to_symbol_map(types::substitution_map &,const
```
Same elsewhere -- just do a global replace.